components/trace/appmap/index.mjs

```diff
@@ -290,12 +305,18 @@ export const compileTrace = (configuration, files, messages, termination) => {
     }
   }
   /* c8 ignore stop */
+
   const appmap = {
     version: VERSION,
     metadata: compileMetadata(configuration, errors, termination),
     classMap: exportClassmap(codebase),
     events: digested_events,
   };
+
+  if (termination.failure) {
+    appmap.metadata.test_failure = makeFailure(termination.failure, codebase);
```

Comment:
I would prefer you change compileMetadata instead of adding this side effect.

Reply:
How is this a side effect? The function is still pure. The reason I did it like this was to avoid the need to pass `codebase` to `compileMetadata`.